pkg/plugins/trivy/plugin.go

```diff
@@ -213,26 +213,55 @@ func (p *plugin) NewConfigForConfigAudit(ctx trivyoperator.PluginContext) (confi
 	return getConfig(ctx)
 }
 
-func (p *plugin) parseImageRef(imageRef, imageDigest string) (v1alpha1.Registry, v1alpha1.Artifact, error) {
-	ref, err := containerimage.ParseReference(imageRef)
+func ParseImageRef(imageRef string, imageDigest string) (v1alpha1.Registry, v1alpha1.Artifact, error) {
+	parts := strings.Split(imageRef, "@")
+	namePart := parts[0]
+	var hasDigest bool
+	var hasTag bool
+	if len(parts) > 1 {
+		hasDigest = true
+	}
+	if len(strings.Split(namePart,":"))>1{ 
+		hasTag = true
+	}
+
+	ref, err := containerimage.ParseReference(namePart, containerimage.WeakValidation)
 	if err != nil {
 		return v1alpha1.Registry{}, v1alpha1.Artifact{}, err
 	}
+
 	registry := v1alpha1.Registry{
 		Server: ref.Context().RegistryStr(),
 	}
 	artifact := v1alpha1.Artifact{
 		Repository: ref.Context().RepositoryStr(),
 	}
-	switch t := ref.(type) {
-	case containerimage.Tag:
-		artifact.Tag = t.TagStr()
-	case containerimage.Digest:
-		artifact.Digest = t.DigestStr()
+
+	tagged, ok := ref.(containerimage.Tag)
+	switch {
+	case ok && hasTag:
+		artifact.Tag = tagged.TagStr()
+	case ok && !hasTag && !hasDigest:
+		artifact.Tag = tagged.TagStr()
+	case ok && hasTag && hasDigest:
+		artifact.Tag = tagged.TagStr()
```

Comment:
`ok` should be `true` for all cases, right? 
and `artifact.Tag = tagged.TagStr()` calls for all cases.
honestly it doesn't look like a good way to check tags/digest.